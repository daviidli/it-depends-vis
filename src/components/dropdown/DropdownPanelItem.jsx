/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'antd';
import { pipe, split, head } from 'ramda';

import './DropdownPanelItem.scss';

const firstParagraph = pipe(split('\n'), head);

const DropdownPanelItem = ({ commit }) => (
	<Row className="dropdown-item" justify="center" align="center" gutter={[16, 8]}>
		<Col span={3}>
			{commit && commit.author && commit.author.avatar_url ? (
				<img
					className="avatar"
					alt="author avatar"
					src={commit.author.avatar_url}
				/>
			) : null}
		</Col>
		<Col span={21}>{firstParagraph(commit.message)}</Col>
	</Row>
);

DropdownPanelItem.propTypes = {
	commit: PropTypes.object.isRequired
};

export default DropdownPanelItem;
