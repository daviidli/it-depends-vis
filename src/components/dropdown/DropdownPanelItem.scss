.avatar {
	object-fit: cover;
	height: 2em;
	width: 2em;
}
