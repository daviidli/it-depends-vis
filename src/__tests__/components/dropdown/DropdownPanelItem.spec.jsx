import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DropdownPanelItem from '../../../components/dropdown/DropdownPanelItem';

const defaultProps = {
	commit: {
		author: {
			avatar_url: 'url'
		},
		message: 'commit msg'
	}
};

const setup = (props = defaultProps) => {
	const wrapper = shallow(<DropdownPanelItem {...props} />);
	return {
		wrapper
	};
};

describe('DropdownPanel component', () => {
	it('should match snapshot', () => {
		const { wrapper } = setup();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should match snapshot without author avatar', () => {
		const props = {
			commit: {
				message: 'commit msg'
			}
		};
		const { wrapper } = setup(props);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
