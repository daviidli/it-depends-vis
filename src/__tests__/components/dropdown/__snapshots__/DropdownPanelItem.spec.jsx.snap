// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownPanel component should match snapshot 1`] = `
<Row
  align="center"
  className="dropdown-item"
  gutter={
    Array [
      16,
      8,
    ]
  }
  justify="center"
>
  <Col
    span={3}
  >
    <img
      alt="author avatar"
      className="avatar"
      src="url"
    />
  </Col>
  <Col
    span={21}
  >
    commit msg
  </Col>
</Row>
`;

exports[`DropdownPanel component should match snapshot without author avatar 1`] = `
<Row
  align="center"
  className="dropdown-item"
  gutter={
    Array [
      16,
      8,
    ]
  }
  justify="center"
>
  <Col
    span={3}
  />
  <Col
    span={21}
  >
    commit msg
  </Col>
</Row>
`;
